boxyhq/jackson
# Feature/add postgres schema support

## What does this PR do?



Fixes #1818 

First commit:
- This commit adds postgres schema support to the app logic. The dev environment uses synchronize function to create tables, and does not run the explicit migrations. We will add schema support for production in the next commit.

Second commit:
- Unlike dev environment, schema will not be created if it does not exist and instead will throw an error. The dev should also make sure that they have sufficient usage and create priviliges on the schema.
- The env variables in .env file will not be picked up by the migration script. Please export them explicitly.
- The migration script inside migration/sql was copied to the postgres directory to accomodate the postgres schema. That migration will now not run for both mssql and postgres.
- You might notice that we set the schema in the dataSource from the env variable and we still modify the SQL strings in migrations file to include schema name. This is because queryRunner when running raw sql commands runs the commands directly without modification. But we still need to set the schema because TypeORM does a select operation on `_jackson_migrations` table and the schema name is needed in the dataSource for that.

## Type of change




## How should this be tested?

I have added tests for the dev environment to cover the first commit. For the second commit,
- create a new schema in your postgres db. Ensure that they have USAGE and CREATE priviliges. A likely error without that would be: [link](https://dba.stackexchange.com/questions/106057/error-no-schema-has-been-selected-to-create-in)
- export DB_URL and POSTGRES_SCHEMA manually so that they are picked up as env variables.
- run migrations for postgres.

Question: 
- .env.example explains what POSTGRES_SCHEMA does. Should it be added to the BoxyHQ online docs. If so, where would be the best place for that?

## Checklist:
